### DISCUSSION TITLE
Question about sending big data [13681]

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#2451] [Q&A]
I find fast-dds 1.9 may loss packet when transmitting ~1MB data within device with best-effort.
All I have tried is,
 - reliable, which may cause big latency;
 - tcp, seems do not work;
 - share memory, have to upgrade to version 2.0;
So is there any other way to fix the problem? Thanks for your help in advance.

**Comment by Hpshboss** [comment]
Did you transmit data which is less than 1MB sucessfully?

**Comment by advancenXX** [comment]
> Did you transmit data which is less than 1MB sucessfully?
Yes, for data less than 64k.

**Comment by Hpshboss** [comment]
I am a newbie for dds, but I give some suggestions:
Firstly, if data is bigger than 64KB, I remember the packet will be seperated into fragment in default, which can handle 8MB steadily for me. Therefore, I think your program's setting may be changed to limiting data size.
Secondly, there are two declaring variable method (IDL or Dynamic). If you use IDL to declare, I think it will be more steadily.

**Comment by advancenXX** [comment]
Thanks, Hpshboss.
I use IDL to declare 1Mb data and there still be packet loss.
For now I think 64k fragment is the reason why I lost data, as there may be 64k packet burst on the Sub side.
